ui/app/components/secret-link.js

```diff
@@ -6,15 +6,41 @@
 import Component from '@glimmer/component';
 import { action } from '@ember/object';
 import { encodePath } from 'vault/utils/path-encoding-helpers';
+import { debug } from '@ember/debug';
 
+/**
+ * @module SecretLink
+ *
+ * @example
+ * ```js
+ * <SecretLink
+ *   @mode="edit"
+ *   @backend="backend-path"
+ *   @secret="my/secret path"
+ *   @queryParams={{hash tab="version"}}
+ *   @disabled={{true}}
+ * />
+ *
+ * @param {string} mode - [required] controls the route link. added to the base route vault.cluster.secrets.backend
```

Comment:
Since params wrapped in brackets `@param {string} [secret]` denotes optional in jsdoc syntax, if you want to be extra explict about things that are required perhaps would be better, I _think_ that will bold it 
```suggestion
 * @param {string} icon - *required* renders the icon for the link
```